src/hash.rs

```diff
@@ -7,7 +7,7 @@ use sha2::{Digest, Sha256};
 use std::fmt;
 
 #[derive(Serialize, Deserialize, Clone, Copy, Default, Eq, PartialEq, Ord, PartialOrd, Hash)]
-pub struct Hash(GenericArray<u8, U32>);
+pub struct Hash(pub GenericArray<u8, U32>);
```

Comment:
What adds a dependency on the internal type? I was hoping to change this to a fixed-size array at some point?

Reply:
That would work as well! It's consumed by the calculate_seed() function which needs access to the byte array of the hash.